js_modules/dagit/packages/core/src/runs/Run.tsx

```diff
@@ -242,7 +258,11 @@ const RunWithData: React.FC<RunWithDataProps> = ({
         newSelected = [filterForExactStep];
 
         // When only one step is selected, set the compute log key as well.
-        setComputeLogKey(stepKey);
+        const matchingLogKey = metadata.logCaptureSteps
+          ? Object.values(metadata.logCaptureSteps).find((info) => info.stepKeys.includes(stepKey))
```

Comment:
It seems like this is one more place the `matchingComputeLogKeyFromStepKey` helper could be used!

Also it looks like we always pass `Object.values(metadata.logCaptureSteps || [])` as the first parameter to the helper. Maybe we move that code into the helper as well to make the callsites look a bit cleaner (eg: just passing `metadata.logCaptureSteps`)